ncm-network/src/main/perl/nmstate.pm

```diff
@@ -121,7 +122,14 @@ sub make_nm_ip_rule
         $thisrule{'ip-to'} = $rule->{to} if $rule->{to};
         $thisrule{'ip-from'} = $rule->{from} if $rule->{from};
         push (@rule_entry, \%thisrule);
+        
+        # Add a default absent rule to match table defined. This will clear any existing rules for this table, instead of merging.
```

Comment:
ah, so this is what i hated about nmstate: the state should reflect the config, so if no rule is defined in the config, there should be no rules.
@aka7 thanks for finding this. but feel free to contact upstream developers and ask them if there is a config option to make nmstate "strict" instead us having to generate this in the config.

Reply:
I have had chat with Redhat and they suggested this. The absent rules can be set as wildcard too.  I've had a discussion with them to see if they'll introduce an option to just replace instead of merge, but will see how far it goes.

for now though, I think this will be ok since we are adding rules/routes per interface and will clear all for that interface.

If you have any other suggestions happy to discuss it.